hyperbahn/advertise_test.go

```diff
@@ -304,6 +304,17 @@ func TestRetryFailure(t *testing.T) {
 	})
 }
 
+func TestCanUnmarshallYAMLFailStrategy(t *testing.T) {
```

Comment:
Can you make this test just go through the different strategies and ensure they come as expected using a table test:

we'll need to add yaml to the godeps (there's no vendoring in this repo, just the godeps.json will be updated).

```go
tests := []struct{
  strategy string
  expected FailStrategy
  wantErr bool
}{
  {
    strategy: "fatal",
    expected: FailStrategyFatal,
  },
  {
    strategy: "ignore",
    expected: FailStrategyIgnore,
  },
  {
    strategy: "unknown",
    wantErr: true,
  },
  {
    strategy: "",
    wantErr: true,
  },
}

for _, tt := range tests {
  var fs FailStrategy
  err := yaml.Unmarshal([]byte(tt.strategy), &fs)
  if tt.wantErr {
    assert.Error(t, err, "Unmarshal %v should fail", tt.strategy)
  } else {
    assert.NoError(t, err, "Unmarshal %v shouldn't fail", tt.strategy)
  }
  if err != nil {
    continue
  }
  assert.Equal(t, tt.expected, fs, "FailStrategy mismatch")
}

```